### ISSUE TITLE
Rust version requirement

### ISSUE THREAD

**Issue by lars-t-hansen** [NordicHPC/sonar#39]
We currently have rust 1.60 as the latest version on some of our HPC systems.  clap 4.3.3 requires anstyle 1.0.0 which requires rust 1.64.  I'll just hack around it for now but in the longer term we should probably figure out what version of rust is both conducive to development velocity and likely to be available where sonar must run.

**Comment by bast** [comment]
Thanks! Good point. Personally I would prefer to not restrict the version because:
- installing the latest version locally in home is relatively easy (this is what I do)
- for running the tool we don't need Rust; we only need it to build

**Comment by lars-t-hansen** [comment]
I can live with that. I'll let you know if there are snags; feel free to close this.

**Comment by lars-t-hansen** [comment]
>     * installing the latest version locally in home is relatively easy (this is what I do)

For posterity:

This strategy has been failing for me on Fox due to what looks like a failure of cargo to spawn a thread and then crashing during rust installation (repeatedly).  There's probably some kind of limit on the Fox login node that is the cause of that, but I don't know yet.  Presumably solvable.

So I tried compiling on my laptop and shipping the binary over, but there's the usual versioning hell: the binary was created against glibc 2.32 but Fox has only the older 2.28 (iirc).

So I tried compiling on my laptop for target=x86_64-unknown-linux-musl, which statically links everything, and that works satisfactorily.  The binary is comparable in size to the dynamically linked code - slightly smaller code, a bit larger data.

There are currently Rust MUSL targets for aarch64 and i686 and armv7 so we should be OK for a while with that strategy, if we need it.  (Nothing yet for riscv but I guess we're not there yet.)
